TombEngine/Objects/TR4/Entity/tr4_demigod.cpp

```diff
@@ -27,6 +27,14 @@ namespace TEN::Entities::TR4
 	constexpr auto DEMIGOD1_WALK_RANGE					   = SQUARE(BLOCK(3));
 	constexpr auto DEMIGOD2_RADIAL_PROJECTILE_ATTACK_RANGE = SQUARE(BLOCK(5));
 	constexpr auto DEMIGOD3_RADIAL_PROJECTILE_ATTACK_RANGE = SQUARE(BLOCK(5));
+	constexpr auto DEMIGOD1_SHOCKWAVE_LIGHT_DURATION = 1.0f;
+
+	struct Demigod1Info
+	{
+		Vector3i LightPosition = Vector3i::Zero;
+	};
+
+	Demigod1Info Demigod1Data;
```

Comment:
Why such complexity? You make a struct which keeps a vector3. You define a global variable which is this struct. Then you refer to this struct only to pass a vector3 from it to `SpawnDynamicLight`. Can't you just directly use this position to spawn it?

Reply:
fixed